File: docs/examples/debug_test.java
public class debug_test {
    public static int factorial(int n) {
        if (n <= 1) return 1;
        return n * factorial(n - 1);
    }
    public static void main(String[] args) {    
        int x = 5;
        int result = factorial(x);
        System.out.println("factorial(" + x + ") = " + result);
    }
}

File: docs/examples/debug_test.pas
program DebugTest;

function Factorial(n: integer): integer;
begin
    if n <= 1 then
        Factorial := 1
    else
        Factorial := n * Factorial(n - 1);
end;

var
    x, result: integer;
begin
    x := 5;
    result := Factorial(x);
    writeln('factorial(', x, ') = ', result);
end.
